In [12]:
import requests

url = "https://www.visitmaryland.org/"

headers = {
    "User-Agent": "Mozilla/5.0 (Windows NT 10.0; Win64; x64)"
}

response = requests.get(url, headers=headers)

print("HTTP Status Code:", response.status_code)


HTTP Status Code: 403


The website returned an HTTP 403 status code, indicating restricted access for automated requests. When extracting visible text, the response contained a bot protection message requesting JavaScript and cookies, confirming that the website blocks non-browser clients

In [14]:
import requests
from bs4 import BeautifulSoup

url = "https://www.visitmaryland.org/"

headers = {
    "User-Agent": "Mozilla/5.0 (Windows NT 10.0; Win64; x64)"
}

response = requests.get(url, headers=headers)

soup = BeautifulSoup(response.text, "html.parser")

# Remove script and style tags
for tag in soup(["script", "style"]):
    tag.decompose()

# Extract visible text
text = soup.get_text(separator="\n", strip=True)

print("\n--- Visible Text from Maryland Page ---\n")
print(text)



--- Visible Text from Maryland Page ---

Just a moment...
Enable JavaScript and cookies to continue


In [15]:
url_nlp = "https://en.wikipedia.org/wiki/Natural_language_processing"
response_nlp = requests.get(url_nlp, headers=headers)

soup_nlp = BeautifulSoup(response_nlp.text, "html.parser")

print("\n--- Headings from Wikipedia NLP Page ---\n")

for tag in ["h1", "h2", "h3"]:
    for heading in soup_nlp.find_all(tag):
        print(heading.get_text(strip=True))



--- Headings from Wikipedia NLP Page ---

Natural language processing
Contents
History
Approaches: Symbolic, statistical, neural networks
Common NLP tasks
General tendencies and (possible) future directions
See also
References
Further reading
External links
Symbolic NLP (1950s – early 1990s)
Statistical NLP (1990s–present)
Statistical approach
Neural networks
Text and speech processing
Morphological analysis
Syntactic analysis
Lexical semantics (of individual words in context)
Relational semantics (semantics of individual sentences)
Discourse (semantics beyond individual sentences)
Higher-level NLP applications
Cognition


In [16]:
print("\n--- All URLs Found on Wikipedia NLP Page ---\n")

links = soup_nlp.find_all("a", href=True)

for link in links[:50]:
    print(link["href"])



--- All URLs Found on Wikipedia NLP Page ---

#bodyContent
/wiki/Main_Page
/wiki/Wikipedia:Contents
/wiki/Portal:Current_events
/wiki/Special:Random
/wiki/Wikipedia:About
//en.wikipedia.org/wiki/Wikipedia:Contact_us
/wiki/Help:Contents
/wiki/Help:Introduction
/wiki/Wikipedia:Community_portal
/wiki/Special:RecentChanges
/wiki/Wikipedia:File_upload_wizard
/wiki/Special:SpecialPages
/wiki/Main_Page
/wiki/Special:Search
https://donate.wikimedia.org/?wmf_source=donate&wmf_medium=sidebar&wmf_campaign=en.wikipedia.org&uselang=en
/w/index.php?title=Special:CreateAccount&returnto=Natural+language+processing
/w/index.php?title=Special:UserLogin&returnto=Natural+language+processing
https://donate.wikimedia.org/?wmf_source=donate&wmf_medium=sidebar&wmf_campaign=en.wikipedia.org&uselang=en
/w/index.php?title=Special:CreateAccount&returnto=Natural+language+processing
/w/index.php?title=Special:UserLogin&returnto=Natural+language+processing
#
#History
#Symbolic_NLP_(1950s_–_early_1990s)
#Statistical

In [17]:
first_paragraph = soup_nlp.find("p").get_text(strip=True)

with open("nlp_intro.txt", "w", encoding="utf-8") as file:
    file.write(first_paragraph)

print("\nFirst paragraph saved to nlp_intro.txt")



First paragraph saved to nlp_intro.txt


In [18]:
print(first_paragraph)

Natural language processing(NLP) is the processing ofnatural languageinformation by acomputer. NLP is a subfield ofcomputer scienceand is closely associated withartificial intelligence. NLP is also related toinformation retrieval,knowledge representation,computational linguistics, andlinguisticsmore broadly.[1]
